### ISSUE TITLE
[Repro]Cityscapes → FoggyCityscapes Results Not Matching Paper (R-34 / R-50), Instance/Global DA Loss Almost Flat

### ISSUE THREAD

**Issue by sparklejojo** [Jeremy-lf/RT-DATR#5]
<h2>Description</h2>
<p>Thank you for releasing this great work and providing the code! I tried to reproduce the <strong>Cityscapes → FoggyCityscapes</strong> experiments (ResNet-34 and ResNet-50), but the results are noticeably lower than those reported in the paper.<br>
To help with debugging, I have attached my <strong>config file</strong> and <strong>training log</strong> below.</p>
<ul>
<li>
<p>The only change I made: set <code inline="">batch_size=8</code> (due to limited resources) and used <strong>2× A6000 GPUs</strong>.</p>
</li>
<li>
<p>Everything else (hyperparameters, training procedure, dataset splits) was kept identical to the paper’s setup.</p>
</li>
<li>
<p>I also plotted the loss curves, and observed that both the <strong>instance loss</strong> and <strong>global DA loss</strong> remain almost flat throughout training. I am not sure if this behavior is expected.</p>
</li>
</ul>
<p>Possible causes I’m considering:</p>
<ol>
<li>
<p>Could this be related to <strong>annotation conversion issues</strong> in Cityscapes/FoggyCityscapes (COCO format/category mapping)?</p>
</li>
<li>
<p>Or could it be a problem with the version mismatch with the paper implementation?</p>
</li>
<li>
<p>Does the change in batch size require corresponding adjustments to <strong>learning rate / iterations / warmup</strong>?</p>
</li>
</ol>
<hr>
<h2>Expected vs. Actual</h2>

Backbone | Setting | Reported (Paper) | Reproduced (My Run)bset
-- | -- | -- | --
ResNet-34 | City → Foggy | 50.4 | 42.6
ResNet-50 | City → Foggy | 52.7 | 44.0 

<hr>
<h2>Training Curve </h2>
<ul>
<li>
<p><code inline="">instance loss</code> and <code inline="">global DA loss</code>: remain nearly flat (very little change).</p>

> ResNet-34

<img width="700" height="700" alt="Image" src="https://github.com/user-attachments/assets/c2533ab2-d0e9-4005-8ec7-cdc94569fc7e" />

> ResNet-50

<img width="700" height="700" alt="Image" src="https://github.com/user-attachments/assets/f6918e56-9ec4-43b3-8022-08ba7c19e8f3" />

</li>
</ul>
<p>Question: is it <strong>normal</strong> for instance/global DA loss to remain almost unchanged on this task? Or does this indicate a setup issue (e.g., gradients not propagating, loss weight too small, etc.)?</p>

<hr>
<h2>What I Need Help With</h2>
<ol>
<li>
<p><strong>Correct config alignment</strong>: With <code inline="">batch_size=8</code> (2× A6000), should I rescale learning rate, iterations, warmup steps, or other hyperparameters?</p>
</li>
<li>
<p><strong>Dataset annotations</strong>: Is there an official COCO conversion script and class mapping list for Cityscapes/FoggyCityscapes?</p>
</li>
<li>
<p><strong>DA loss behavior</strong>: Should <code inline="">instance/global DA loss</code> decrease significantly during training, or is a flat curve expected?</p>
</li>
<li>
<p><strong>Version alignment</strong>: Could you provide the release corresponding to the paper experiments?</p>
</li>
<li>
<p><strong>Eval details</strong>: Any special filtering of categories or evaluation scripts (e.g., only 8 classes, IoU specifics) that might affect results?</p>
</li>
</ol>
<hr>
<h2>Attachments</h2>

[da_rtdetr_r34_backbone_encoder_instance_dn_cmt_city2foggycity.yml](https://github.com/user-attachments/files/22000421/da_rtdetr_r34_backbone_encoder_instance_dn_cmt_city2foggycity.yml)
[train_cityscapes2foggycityscapes_r34.log](https://github.com/user-attachments/files/22000423/train_cityscapes2foggycityscapes_r34.log)

[da_r50_rtdetr_backbone_encoder_instance_dn_cmt_city2foggycity.yml](https://github.com/user-attachments/files/22000424/da_r50_rtdetr_backbone_encoder_instance_dn_cmt_city2foggycity.yml)
[train_cityscapes2foggycityscapes_r50.log](https://github.com/user-attachments/files/22000422/train_cityscapes2foggycityscapes_r50.log)

</li>
</ul>

<p>Thanks again for your excellent work! Looking forward to your guidance.</p>

**Comment by Jeremy-lf** [comment]
Thank you for following our work. I have briefly verified your configuration file and found that your pre-training usage is incorrect. In the use of domain adaptation methods, in order to ensure the effectiveness of pseudo labels, we usually treat the model trained on the source domain data as initialization, rather than simply using the backbone network for pre training.

**Comment by Jeremy-lf** [comment]
Thank you for following our work. I have briefly verified your configuration file and found that your pre-training usage is incorrect. In the use of domain adaptation methods, in order to ensure the effectiveness of pseudo labels, we usually treat the model trained on the source domain data as initialization, rather than simply using the backbone  pre-training.

```
pretrain_weights: "/home/qtt/.cache/paddle/weights/ResNet34_vd_pretrained.pdparams"
```
changed like this:
```
pretrain_weights: "source data trained model"
```

**Comment by sparklejojo** [comment]
Thanks a lot for the quick reply and for sharing this great work!

To double-confirm my understanding of the ` trained on source domain data `  used as `pretrain_weights` for UDA:

**Which config should we use for Stage-1 (source-only, Cityscapes→Cityscapes) to produce the checkpoint that initializes Stage-2 UDA?**

* **Option A (plain RT-DETR for source-only):**
  Use the base RT-DETR config, e.g.

  ```
  configs/rtdetr/rtdetr_r34vd_6x_coco.yml
  ```

  and switch datasets to Cityscapes (train/val), pick the best Cityscapes-val checkpoint, and then use it as:

  ```
  pretrain_weights: /path/to/stage1_best_model.pdparams
  ```

  when running

  ```
  configs/domain_adaption/da_rtdetr_r34_backbone_encoder_instance_dn_cmt_city2foggycity.yml
  ```

* **Option B (DA config):**
  Start from

  ```
  configs/domain_adaption/da_rtdetr_r34_backbone_encoder_instance_dn_cmt_city2foggycity.yml
  ```

  but train only on Cityscapes train and validate on Cityscapes val to get the best checkpoint. Then switch FoggyCityscapes val for Stage-2.

1. Could you please confirm **which option you used in the paper/released repo**?
I tried Option B, Stage-1 (pure supervised, City→City): best AP50 = 51.4 on Cityscapes-val. Stage-2 (City→Foggy): best AP50 = 46.6 on Foggy-val., which is still below 50.4. 

2. Also, during Stage-2 all the DA losses change very little throughout training.  Is this expected? If not, what would you suggest checking?
<img width="600" height="600" alt="Image" src="https://github.com/user-attachments/assets/649b7760-a97d-4f11-ad00-7f64d223d82a" />

[train_20250828_092441.log](https://github.com/user-attachments/files/22026486/train_20250828_092441.log)


Thanks again for your time and guidance!

**Comment by Jeremy-lf** [comment]
1. Option A is right.
2. Under normal case, the loss value decreases slowly, and I suspect it may be caused by pre training
